src/main/java/org/openrewrite/staticanalysis/SimplifyConstantTernaryExecution.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.Repeat;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+import java.time.Duration;
+
+public class SimplifyConstantTernaryExecution extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Simplify constant ternary branch execution";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Checks for ternary expressions that are always `true` or `false` and simplifies them.";
```

Comment:
Can we add this to the SimplifyTernary Refaster recipe?

Reply:
Not if I want to include the call to `cleanupBooleanExpression`

Reply:
I think if we were to make sure the existing `SimplifyBooleanExpressionVisitor` can handle this case, then we should be fine, as we have the dedicated `SIMPLIFY_BOOLEANS` embedding option for that.

Reply:
Picked up in
- https://github.com/openrewrite/rewrite/pull/4617